From e4442a217f4222db2927cda0726f43085bb8efe0 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 18 Aug 2014 19:56:53 -0400
Subject: [scsi] qla2xxx: Add MBC option for fast SFP data access

Message-id: <1408391817-6389-66-git-send-email-cdupuis@redhat.com>
Patchwork-id: 87963
O-Subject: [PATCH RHEL 7.1 e-stor 65/69] qla2xxx: Add MBC option for fast SFP data access.
Bugzilla: 1089346
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Joe Carnuccio <joe.carnuccio@qlogic.com>

scsi commit id 51e748838c6a31cd1f6b1333517b2ea883d57488

Signed-off-by: Joe Carnuccio <joe.carnuccio@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_attr.c b/drivers/scsi/qla2xxx/qla_attr.c
index 356b99e..b261542 100644
--- a/drivers/scsi/qla2xxx/qla_attr.c
+++ b/drivers/scsi/qla2xxx/qla_attr.c
@@ -578,7 +578,7 @@ do_read:
   }
 
   rval = qla2x00_read_sfp(vha, ha->sfp_data_dma, ha->sfp_data,
-      addr, offset, SFP_BLOCK_SIZE, 0);
+      addr, offset, SFP_BLOCK_SIZE, BIT_1);
   if (rval != QLA_SUCCESS) {
    ql_log(ql_log_warn, vha, 0x706d,
        "Unable to read SFP data (%x/%x/%x).\n", rval,
-- 
1.7.1